pkg/reboot/calculator.go

```diff
@@ -106,14 +115,79 @@ func (s *safeTimeCalculator) calcMinTimeAssumeRebooted() error {
 	s.log.Info("calculated minTimeToAssumeNodeRebooted is:", "minTimeToAssumeNodeRebooted", minTime)
 	s.minTimeToAssumeNodeRebooted = minTime
 
-	if s.timeToAssumeNodeRebooted < minTime {
-		err := fmt.Errorf("snr agent can't start: the requested value for SafeTimeToAssumeNodeRebootedSeconds is too low")
-		s.log.Error(err, err.Error(), "requested SafeTimeToAssumeNodeRebootedSeconds", s.timeToAssumeNodeRebooted, "minimal calculated value for SafeTimeToAssumeNodeRebootedSeconds", minTime)
+	//update related logic of min time on configuration if necessary
+	if err := s.manageSafeRebootTimeInConfiguration(minTime); err != nil {
 		return err
 	}
+
 	return nil
 }
 
+// manageSafeRebootTimeInConfiguration does two things:
+//  1. It sets Status.MinSafeTimeToAssumeNodeRebootedSeconds in case it's changed by latest calculation.
+//  2. It Adds/removes config.Status.SpecSafeTimeOverriddenWarning if necessary.
+func (s *safeTimeCalculator) manageSafeRebootTimeInConfiguration(minTime time.Duration) error {
+	minTimeSec := int(minTime.Seconds())
+	config, err := s.getConfig()
+	if err != nil {
+		return err
+	}
+	orgConfig := config.DeepCopy()
+	prevMinRebootTimeSec := config.Status.MinSafeTimeToAssumeNodeRebootedSeconds
+	if prevMinRebootTimeSec != minTimeSec {
+		config.Status.MinSafeTimeToAssumeNodeRebootedSeconds = minTimeSec
+	}
+	//Manage warning
+	config.Status.SpecSafeTimeOverriddenWarning = ""
+	if config.Spec.SafeTimeToAssumeNodeRebootedSeconds > 0 && minTimeSec > config.Spec.SafeTimeToAssumeNodeRebootedSeconds {
+		config.Status.SpecSafeTimeOverriddenWarning = v1alpha1.SafeTimeToAssumeNodeRebootedSecondsWarning
+		s.log.Info("SafeTimeToAssumeNodeRebootedSeconds is overridden by calculated value because it's invalid, calculated value stored in Status.MinSafeTimeToAssumeNodeRebootedSeconds would be used instead")
+		events.WarningEvent(s.recorder, config, "SafeTimeToAssumeNodeRebootedSecondsInvalid", "SafeTimeToAssumeNodeRebootedSeconds is overridden by calculated value")
+	}
+
+	if !reflect.DeepEqual(config, orgConfig) {
```

Comment:
>I experienced issues with DeepEqual for this, that's why I do it different in NHC

Thanks, I've used your warning to read a bit about common pitfalls of DeepEqual, I'm not aware of any issues for this use case.
Am I missing something? also in case you remember it would be great if you can explain what was the issue in NHC.  

>you know if changed something, not?

Indeed I do have the potential data to know if something changed, however I think it simplify the code to just assign the values according to the current state and than compare end state vs initial state over multiple checks in every step of the way.

Reply:
>  in case you remember

unfortunately I don't 🤔